File: backend/app/models/study_plan.py
from datetime import date
from typing import List

from pydantic import BaseModel

from .subject import Subject


class StudyPlan(BaseModel):
    """
    High-level study plan for a user.

    Links a user to a set of subjects and a target exam date.
    """

    user_id: int
    subjects: List[Subject]
    start_date: date
    target_exam_date: date

File: mobile_app/upsc_app/lib/models/study_plan.dart
import 'subject.dart';

/// Placeholder study plan model for the Flutter app.
///
/// Mirrors the core fields from the backend `StudyPlan` model.
class StudyPlan {
  final int userId;
  final List<Subject> subjects;
  final DateTime startDate;
  final DateTime targetExamDate;

  StudyPlan({
    required this.userId,
    required this.subjects,
    required this.startDate,
    required this.targetExamDate,
  });
}

